internal/dev_server/ui/src/Flags.tsx

```diff
@@ -231,6 +243,7 @@ function Flags({
                   setSearchTerm(e.target.value);
                   setCurrentPage(0); // Reset pagination
                 }}
+                value={searchTerm}
```

Comment:
Previously the input of the search box wasn't actually coupled to the display of the search... We were just relying on onChange events.